Include required wxMac library into Mac binary package

It could be hard for end-user to manually install wxWidgets binaries on Mac (using brew) required to run MMEX.

We should statically link this library with MMEX for Mac or include shared libraries copy into distributed Mac package.

While static linking MMEX with libpng it gives following error on OSX>10.11:
```
Undefined symbols for architecture x86_64:
  "_inflateValidate", referenced from:
      _png_inflate_claim in libpng.a(pngrutil.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Preinstalled zlib cannot be linked with libpng even if libpng is compiled locally... I'll try to recompile zlib.

--- a/.travis.yml
+++ b/.travis.yml
@@ -33,9 +33,10 @@ env:
 matrix:
   fast_finish: true
   include: # add non-linux targets
-  - { os: osx, osx_image: xcode6.4, env: OS=osx DIST=yosemite }
-  - { os: osx, osx_image: xcode8, env: OS=osx DIST=el_capitan }
-  - { os: osx, osx_image: xcode9, env: OS=macos DIST=sierra }
+  - { os: osx, osx_image: xcode6.4, env: OS=osx DIST=10.10 }
+  - { os: osx, osx_image: xcode8,   env: OS=osx DIST=10.11 }
+  - { os: osx, osx_image: xcode9.1, env: OS=macos DIST=10.12 }
+  - { os: osx, osx_image: xcode9.1, env: OS=macos DIST=10.13 }
   exclude:
   - { osx_image: xcode6.4, env: BUILD_TESTS=ON } # libtoolize problems
   allow_failures:
@@ -53,8 +54,9 @@ matrix:
   - env: OS=opensuse DIST=42.3
   - env: OS=archlinux
   - env: OS=linuxmint DIST=serena
-  - env: OS=osx DIST=yosemite
-  - env: OS=osx DIST=el_capitan
+  - env: OS=osx DIST=10.10
+  - env: OS=osx DIST=10.11
+  - env: OS=macos DIST=10.12
   - env: OS=slackware DIST=14.2
 
 install: |-
@@ -65,12 +67,12 @@ install: |-
       docker build dockers/$OS${DIST:+.$DIST} --cache-from ${DOCKER} -t ${DOCKER}
       ;;
     osx)
+      MACOSX_DEPLOYMENT_TARGET=$DIST
       brew update # see travis-ci/travis-ci#8552
-      BREW=wxmac
-      brew list gettext >/dev/null || BREW="$BREW gettext"
-      brew list libtool >/dev/null || BREW="$BREW libtool"
-      brew list automake >/dev/null || BREW="$BREW automake"
-      brew install $BREW && brew link --force gettext libtool
+      for p in gettext libtool automake zlib;
+        do brew list $p >/dev/null || brew install --verbose $p; done
+      brew link --force gettext libtool zlib
+      brew install --verbose wxmac --with-static --with-stl
       ;;
   esac
   mkdir build && cd build
@@ -96,7 +98,7 @@ notifications: # set notification options
 after_success:
 # Rename mac packages
 - if [ $TRAVIS_OS_NAME = osx ]; then
-    for p in *.dmg; do mv -v $p ${p%Darwin.*}$OS.$DIST.dmg; done; fi
+    for p in *.dmg; do mv -v $p ${p%Darwin.*}$OS$DIST.dmg; done; fi
 # Rename Slackware packages
 - if [ $OS = slackware ]; then
     for p in *.tar.xz; do mv -v $p ${p%.tar.xz}.txz; done; fi

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,12 +179,26 @@ ENDIF()
 SET(wxWidgets_USE_REL_AND_DBG ON)
 IF(WIN32)
     #wxScintilla is needed on Windows
-    FIND_PACKAGE(wxWidgets REQUIRED core qa net html xml aui adv stc webview base scintilla REQUIRED)
+    FIND_PACKAGE(wxWidgets REQUIRED
+        core qa net html xml aui adv stc webview base scintilla REQUIRED)
 ELSE()
-    FIND_PACKAGE(wxWidgets REQUIRED core qa net html xml aui adv stc webview base REQUIRED)
+    FIND_PACKAGE(wxWidgets REQUIRED
+        core qa net html xml aui adv stc webview base REQUIRED)
 ENDIF()
 IF(wxWidgets_VERSION_STRING AND wxWidgets_VERSION_STRING VERSION_LESS 2.9.2)
     MESSAGE(SEND_ERROR "wxWidget version to old.")
+ENDIF()
+IF(APPLE AND wxWidgets_LIBRARIES MATCHES "${CMAKE_STATIC_LIBRARY_PREFIX}wx_baseu-[0-9\\.]+${CMAKE_STATIC_LIBRARY_SUFFIX}")
+    # use static libs in place of dynamic if using static wxMac
+    FOREACH(deplib png jpeg tiff z)
+        FIND_LIBRARY(${deplib}path ${CMAKE_STATIC_LIBRARY_PREFIX}${deplib}${CMAKE_STATIC_LIBRARY_SUFFIX})
+        IF(${deplib}path)
+            STRING(REPLACE "-l${deplib}" "${${deplib}path}"
+                wxWidgets_LIBRARIES "${wxWidgets_LIBRARIES}")
+        ENDIF()
+    ENDFOREACH()
+    UNSET(deplib)
+    UNSET(${deplib}path)
 ENDIF()
 ADD_LIBRARY(wxWidgets INTERFACE)
 TARGET_INCLUDE_DIRECTORIES(wxWidgets INTERFACE ${wxWidgets_INCLUDE_DIRS})

